src/commands/run.rs

```diff
@@ -78,45 +77,53 @@ pub async fn handle_run(args: RunArgs) -> Result<()> {
     tracing::info!("Starting slight");
 
     let mut host_builder = build_store_instance(&toml, &args.slightfile, &args.module).await?;
-    if let Some(io_redirects) = args.io_redirects {
+    if let Some(io_redirects) = args.io_redirects.clone() {
         tracing::info!("slight io redirects were specified");
         host_builder = host_builder.set_io(io_redirects);
     }
     let (mut store, instance) = host_builder.build().await;
 
     let caps = toml.capability.as_ref().unwrap();
-
+    let http_enabled;
+
+    if toml.specversion == "0.1" {
+        http_enabled = caps.iter().any(|cap| cap.name == "http");
+    } else if toml.specversion == "0.2" {
+        http_enabled = caps
+            .iter()
+            .any(|cap| cap.resource.as_ref().expect("missing resource field") == "http");
+    } else {
+        bail!("unsupported toml spec version");
+    }
     // looking for the http capability.
-    let http_enabled = if cfg!(feature = "http-server") {
-        let http_enabled;
+    if cfg!(feature = "http-server") && http_enabled {
+        log::debug!("Http capability enabled");
+        update_http_states(
```

Comment:
@Mossaka I needed to add the io_redirects to the HTTP instance builder as well as the host_builder.

Reply:
sounds good!